pypesto/objective/amici_objective.py

```diff
@@ -429,9 +427,12 @@ def apply_steadystate_guess(self, condition_ix, x):
             if guess_data['x_ss'] is not None:
                 x_ss_guess = guess_data['x_ss']
             if guess_data['sx_ss'] is not None:
-                x_ss_guess += guess_data['sx_ss'].transpose().dot(
+                linear_update = guess_data['sx_ss'].transpose().dot(
                     (x_sim - guess_data['x'])
                 )
+                # limit linear updates to max 20 % elementwise change
+                if (x_ss_guess/linear_update).max() < 0.2:
```

Comment:
is there a reason why 20%? that looks rather arbitrary, but if it works better then ok.

Reply:
its arbitrary, we do not have access to second order terms so we cannot judge the approximation quality. This definitely also ensures that initial states remain positive ;)